test/unit/com/unifina/service/CanvasServiceSpec.groovy

```diff
@@ -362,6 +363,23 @@ class CanvasServiceSpec extends BeanMockingSpecification {
 		thrown(InvalidStateException)
 	}
 
+	def "updateExisting lets save canvas in invalid state"() {
+		setup:
+		service.signalPathService = Mock(SignalPathService)
+		def cmd = new SaveCanvasCommand(
+			name: "canvas",
+			modules: [],
+			settings: [:]
+		)
+
+		when:
+		service.updateExisting(myFirstCanvas, cmd, me)
+
+		then:
+		1 * service.signalPathService.reconstruct(_, _) >> { throw new ModuleException("mocked", null, null) }
+		thrown ModuleException
```

Comment:
should check here that the canvas fields (`name`, `json`, etc.) have been updated!